bool[] sonic_too_close
